model/host/db.go

```diff
@@ -325,13 +325,19 @@ func NeedsNewAgent(currentTime time.Time) db.Q {
 	})
 }
 
-func RemoveAllStaleInitializing() error {
-	return db.RemoveAll(Collection,
-		bson.M{
-			StatusKey:     evergreen.HostUninitialized,
-			UserHostKey:   false,
-			CreateTimeKey: bson.M{"$lt": time.Now().Add(-3 * time.Minute)},
-		})
+func RemoveStaleInitializing(distroID string) error {
```

Comment:
The behavior of this function is not obvious from its name and signature. Add a function comment that passing the empty string causes it to remove stale initializing across all distros.